cftime/src/parser.rs

```diff
@@ -0,0 +1,391 @@
+            }
+            Calendars::AllLeap | Calendars::Day366 => {
+                Some(CFDates::DateAllLeap(DateAllLeap::new(year, month, day)))
+            }
+            Calendars::Julian => None,
+            _ => None,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum CFDatetimes {
+    DateTimeProlepticGregorian(DateTimeProlepticGregorian),
+    DateTimeAllLeap(DateTimeAllLeap),
+    DateTimeNoLeap(DateTimeNoLeap),
+    DateTime360Day(DateTime360Day),
+    DateTimeJulian(DateTimeJulian),
+}
+struct CFDateTimeFactory {}
+
+impl CFDateTimeFactory {
+    fn build(&self, date: CFDates, time: Time, tz: Tz) -> Option<CFDatetimes> {
+        match date {
+            CFDates::DateProlepticGregorian(date) => Some(CFDatetimes::DateTimeProlepticGregorian(
+                DateTimeProlepticGregorian::new(date, time, tz),
+            )),
+
+            CFDates::Date360Day(date) => Some(CFDatetimes::DateTime360Day(DateTime360Day::new(
+                date, time, tz,
+            ))),
+            CFDates::DateNoLeap(date) => Some(CFDatetimes::DateTimeNoLeap(DateTimeNoLeap::new(
+                date, time, tz,
+            ))),
+            CFDates::DateAllLeap(date) => Some(CFDatetimes::DateTimeAllLeap(DateTimeAllLeap::new(
+                date, time, tz,
+            ))),
+            CFDates::DateJulian(date) => None,
+            _ => None,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ParsedCFTime {
+    pub duration: CFDuration,
+    pub from: CFDatetimes,
+}
+
+/// Parse a CF compatible string into two components
+pub fn cf_parser(input: &str, calendar: Option<Calendars>) -> Result<ParsedCFTime, ParseError> {
+    let since = tuple((space1, tag("since"), space1));
+    let calendar = match calendar {
+        Some(calendar) => calendar,
+        None => Calendars::default(),
+    };
+    let x = all_consuming(separated_pair(
+        |x| duration(x, &calendar),
+        since,
+        |x| datetime(x, &calendar),
+    ))(input)
+    .map(|(_, o)| ParsedCFTime {
+        duration: o.0,
+        from: o.1,
+    })
+    .map_err(|e| ParseError(format!("{}", e)));
+    x
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    fn parse(input: &str) {
+        println!("{:?}", cf_parser(input, None).unwrap())
+    }
+
+    #[test]
+    fn cf_conventions_document() {
+        parse("days since 1990-1-1 0:0:0");
+        parse("seconds since 1992-10-8 15:15:42.5 -6:00");
+        parse("days since 1-7-15 0:0:0");
+        parse("days since 1-1-1 0:0:0");
+    }
+
+    #[test]
+    fn cftime_py_setup() {
+        parse("hours since 0001-01-01 00:00:00");
+        parse("hours since 0001-01-01 00:00:00");
+        parse("hours since 0001-01-01 00:00:00 -06:00");
+        parse("seconds since 0001-01-01 00:00:00");
+        parse("days since 1600-02-28 00:00:00");
+        parse("days since 1600-02-29 00:00:00");
+        // parse("days since 1600-02-30 00:00:00");
```

Comment:
This date can exists in other calendars, just not the default

Reply:
We should try parsing as different calendars

Reply:
I agree, i did not test for other calendars yet, but i'll do